### PR TITLE
[modeline] move modeline features to separated file

### PR DESCRIPTION
**Pull request by ericdallo**
- Move modeline code actions to `lsp-modeline.el`
- Move modeline diagnostics to `lsp-modeline.el`
- Rename all `lsp-diagnostics-modeline-*` -> `lsp-modeline-diagnostics-*` with `define-obsole-*`
- Add `lsp-modeline-diagnostics-mode` to `lsp-auto-configure` if the new flag `lsp-modeline-diagnostics-enable` is  `t`.

### CODE REVIEW

**Comment by ericdallo** [comment]
I'm not sure why the [tests are failing](https://github.com/emacs-lsp/lsp-mode/pull/1931/checks?check_run_id=882250124)

File: lsp-modeline.el [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+    (add-hook 'lsp-on-idle-hook 'lsp-modeline--check-code-actions nil t)
+    (add-hook 'lsp-configure-hook #'lsp-modeline--enable-code-actions nil t)
+    (add-hook 'lsp-unconfigure-hook #'lsp-modeline--disable-code-actions nil t))
+   (t
+    (remove-hook 'lsp-on-idle-hook 'lsp-modeline--check-code-actions t)
+    (remove-hook 'lsp-configure-hook #'lsp-modeline--enable-code-actions t)
+    (remove-hook 'lsp-unconfigure-hook #'lsp-modeline--disable-code-actions t)
+    (setq global-mode-string (remove '(t (:eval lsp-modeline--code-actions-string)) global-mode-string)))))
+
```
Comment (yyoncho) [inline lsp-modeline.el:155]: too much space.

Comment (ericdallo) [reply lsp-modeline.el:155]: It's the page break: 
image

Comment (yyoncho) [reply lsp-modeline.el:155]: Ah, ok.

**Pull request merged**